internal/user/user.go

```diff
@@ -0,0 +1,62 @@
+package user
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/saucelabs/saucectl/internal/credentials"
+	"github.com/saucelabs/saucectl/internal/requesth"
+)
+
+// Client service
+type Client struct {
+	HTTPClient  *http.Client
+	URL         string
+	Credentials credentials.Credentials
+}
+
+// User represents user data structure
+type User struct {
+	ID           string       `json:"id"`
+	Groups       []Group      `json:"groups"`
+	Organization Organization `json:"organization"`
+}
+
+// Group represents the group that the user belongs to
+type Group struct {
+	ID string `json:"id"`
+}
+
+// Organization represents the organization that the user belongs to
+type Organization struct {
+	ID string `json:"id"`
+}
+
+// Get user data
+func (c *Client) Get(ctx context.Context) (User, error) {
```

Comment:
The user/IAM domain shouldn't be tied to this particular implementation.

I'd recommend creating an `iam` package that defines User, Group and Organization, as well as an interface for retrieving said user data.

Anything using this code is effectively not testable, because you can't substitute its implementation for anything else.

Reply:
👌